recipes/ftjam/all/conanfile.py

```diff
@@ -1,127 +1,140 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
 import os
 
-required_conan_version = ">=1.33.0"
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import cross_building
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, chdir, copy, export_conandata_patches, get, load, replace_in_file, save
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc, NMakeToolchain
+
+required_conan_version = ">=1.47.0"
 
 
 class FtjamConan(ConanFile):
     name = "ftjam"
     description = "Jam (ftjam) is a small open-source build tool that can be used as a replacement for Make."
-    topics = ("conan", "ftjam", "build", "make")
+    license = "BSD-3-Clause"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.freetype.org/jam/"
-    license = "BSD-3-Clause"
-    exports_sources = "patches/*"
-    settings = "os", "arch", "compiler", "build_type"
+    topics = ("build", "make")
 
-    _autotools = None
+    package_type = "application"
+    settings = "os", "arch", "compiler", "build_type"
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def configure(self):
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
-    def validate(self):
-        if self.settings.compiler == "Visual Studio":
-            raise ConanInvalidConfiguration("ftjam doesn't build with Visual Studio yet")
-        if hasattr(self, "settings_build") and tools.cross_building(self):
-            raise ConanInvalidConfiguration("ftjam can't be cross-built")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def package_id(self):
         del self.info.settings.compiler
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+    def validate(self):
+        if is_msvc(self):
+            # Build fails with
+            # NMAKE : fatal error U1077: 'jam0 JamFile' : return code '0xc0000005'
+            raise ConanInvalidConfiguration("ftjam doesn't build with Visual Studio yet")
+        if hasattr(self, "settings_build") and cross_building(self):
+            raise ConanInvalidConfiguration("ftjam can't be cross-built")
```

Comment:
```suggestion
            raise ConanInvalidConfiguration("ftjam doesn't build with Visual Studio yet")

    def validate_build(self):    
        if hasattr(self, "settings_build") and cross_building(self):
            raise ConanInvalidConfiguration("ftjam can't be cross-built")
```

Cross building checks should be done in the `validate_build` method. Historically CCI recipes were plaicng them in the `validate` method because the CI lacked the logic to handle `validate_build` - but this is no longer the case

(The case where a configuration is disabled because it does not build on it because of recipe reasons is a bit different, I still need to come to a conclusion on that one, but is fine for me for now)

Reply:
Oh, good to know. Thanks!